pkg/lib/authenticationflow/declarative/intent_signup_flow_step_identify.go

```diff
@@ -74,7 +74,7 @@ func (i *IntentSignupFlowStepIdentify) GetVerifiableClaims(_ context.Context, _
 		return nil, nil
 	}
 
-	m, ok := authflow.FindMilestoneInCurrentFlow[MilestoneDoCreateIdentity](flows.Nearest)
+	m, ok := authflow.FindFirstMilestone[MilestoneDoCreateIdentity](flows.Nearest)
```

Comment:
It seems using milestone with nested steps could easily cause bugs, I am thinking about should we redefine milestone as "node / intent created by this intent". i.e. We always search in flows.Nearest only.

Reply:
Sure. I think we need to review the existing usage of milestone, and classify each usage into different groups. If we only have a group ("node / intent created by this intent"), then we can change the behavior.

Reply:
Created an issue https://linear.app/authgear/issue/DEV-1245/revise-usage-of-milestones-in-authflow for this